src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -134,14 +134,13 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
             IList<LockFileTarget> targetsList = null;
             T installedPackages;
             T transitivePackages = default;
+            Dictionary<string, TransitiveEntry> transitiveOrigins = default;
             if (IsInstalledAndTransitiveComputationNeeded)
             {
                 // clear the transitive packages cache, since we don't know when a dependency has been removed
                 installedPackages = new T();
-                if (includeTransitivePackages)
-                {
-                    transitivePackages = new T();
-                }
+                transitivePackages = new T();
```

Comment:
On [line 165 before your fix](https://github.com/NuGet/NuGet.Client/blob/87fc99d5049cfa8fd23795e96d30c78aace32cb6/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs#L165): there's another time where the transitivePackages cache is reset. Should the transitiveOrigins be reset there too? If not, what is different about that case?

Reply:
Good catch!